### DISCUSSION TITLE
`@CasePathable` Support annotate on Extension

### DISCUSSION THREAD

**Discussion by AFutureD** [pointfreeco/swift-case-paths#135] [Ideas]
An `action` or any other enum should be independent, and the support for CasePath should be unrelated.
CasePath can be supported through an extension.

Such as:
```swift
@CasePathable
enum ContactsFeatureAction {
    case addButtonTapped
    case deleteButtonTapped(id: Contact.ID)
    case present(PresentationAction<ContactsFeature.Destination.Action>)
    case contactOnPath(StackAction<ContactDetailFeature.State, ContactDetailFeature.Action>)
}
```

It may support things like this, or other ways.
```swift
enum ContactsFeatureAction {
    case addButtonTapped
    case deleteButtonTapped(id: Contact.ID)
    case present(PresentationAction<ContactsFeature.Destination.Action>)
    case contactOnPath(StackAction<ContactDetailFeature.State, ContactDetailFeature.Action>)
}

@CasePathable
extension ContactsFeatureAction: CasePaths.CasePathable {}
```

**Comment by stephencelis** [comment]
@AFutureD This unfortunately isn't possible with how macros work in Swift. Macros can only "see" code that they are attached to, so there's no way for the `@CasePathable extension` to see the enum's cases and generate the necessary code.

There is some talk about giving macros access to some parent scope info in the future, but it's not clear if it would help in this case. You may want to start a discussion on the Swift forums, or open a feature request if you'd like to see this ability in the language.

**Reply by AFutureD** [reply]
Thank you.
I have tried it too. It's impossible to get enum implementation or pass enum Type to annotation, even using "runtime method".
